### ISSUE TITLE
Fix nesting-selector-no-missing-scoping-root for CSS-in-JS

### ISSUE THREAD

**Issue by tu4mo** [stylelint/stylelint#8727]
### What minimal example or steps are needed to reproduce the bug?

```
const Root = styled.div`
  &::before {
    color: red;
  }
`;
```

### What minimal configuration is needed to reproduce the bug?

```
{
  "nesting-selector-no-missing-scoping-root": true
}
```

### How did you run Stylelint?

VS Code extension.

### Which Stylelint-related dependencies are you using?

```
    "stylelint": "16.23.1",
    "stylelint-config-recommended": "17.0.0",
```

### What did you expect to happen?

No error.

### What actually happened?

Unexpected missing scoping root (nesting-selector-no-missing-scoping-root)

**Comment by jeddy3** [comment]
@tu4mo Thanks for the report and for using the template. I can reproduce it [using the demo](https://stylelint.io/demo/#N4Igxg9gJgpiBc4IDsDOAXABAJQhLAvJhgJ4A2MUAdFAJYBuABgDrKaYBk88ARjAGYQATjEzBW7dpDLD4mEVADcEzAF9WjRSAA0IfrQoA5AIYBbOIhgAPMwAcKVMKlQ6kyfQHMEIcW0zMQa3QYZChUALkAbRV2ANIKMlpkdABaSHdaDxSMY1DjISgAlQBdbRUAsABXDAhTAGUSZOMrCP8QWwgMJ1Rs9HJKbMb0ZqLkVVd0zwAxYVNjdG8AK1QUV1hbF0RfWJB4mETk1oCyeZgMALK-OL6EpNTJzN7cqHzChDaT4POdcvbO9G6vX6UEGTRa72Op2+rHGqiAA).

We can use the [`isInDocument`](https://github.com/stylelint/stylelint/blob/main/lib/utils/isInDocument.mjs) util added https://github.com/stylelint/stylelint/pull/8701 to fix the false positive in favour of a false negative (, which we can address as part of https://github.com/stylelint/stylelint/issues/8719 and https://github.com/stylelint/stylelint/issues/6833).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by karlhorky** [comment]
## Workaround

Until this is fixed, a workaround is to disable `nesting-selector-no-missing-scoping-root` for CSS-in-JS files, eg. if you're using [`postcss-styled-syntax`](https://www.npmjs.com/package/postcss-styled-syntax):

`stylelint.config.js`

```js
/** @type {import('stylelint').Config} */
const config = {
  extends: ['stylelint-config-recommended'],
  overrides: [
    {
      files: [
        '**/*.js',
        '**/*.cjs',
        '**/*.mjs',
        '**/*.jsx',
        '**/*.ts',
        '**/*.tsx',
      ],
      customSyntax: 'postcss-styled-syntax',
      rules: {
        // TODO: Remove this configuration once
        // nesting-selector-no-missing-scoping-root no longer
        // causes false positives for CSS-in-JS
        // - https://github.com/stylelint/stylelint/issues/8727#issuecomment-3199832744
        'nesting-selector-no-missing-scoping-root': null,
      },
    },
  ],
};

export default config;
```

**Comment by jeddy3** [comment]
Closed by https://github.com/stylelint/stylelint/pull/8905
